api/src/main/java/org/apache/iceberg/expressions/ExpressionUtil.java

```diff
@@ -158,8 +164,7 @@ public <T> Expression predicate(UnboundPredicate<T> pred) {
           return new UnboundPredicate<>(pred.op(), pred.term(), (T) sanitize(pred.literal()));
         case IN:
         case NOT_IN:
-          Iterable<String> iter =
-              () -> pred.literals().stream().map(ExpressionUtil::sanitize).iterator();
+          Iterable<String> iter = () -> sanitizeValuesList(pred.literals()).iterator();
```

Comment:
I think that this could be done in `StringSanitizer`, but probably shouldn't be done in this sanitizer. The string sanitizer should be what's used for log messages where this makes sense. But the expression sanitizer is intended to pass expressions to metrics collection, where we want to be able to estimate the cardinality of literals.

Reply:
reverting this one